src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -425,16 +421,31 @@ internal unsafe SocketError DoOperationReceiveMultiBuffer(SafeSocketHandle handl
             NativeOverlapped* overlapped = AllocateNativeOverlapped();
             try
             {
+                int count = _bufferListInternal!.Count;
+                WSABuffer[]? array = null;
+                if (count > MaxStackWSABuffers)
+                {
+                    array = ArrayPool<WSABuffer>.Shared.Rent(count);
+                }
+
+                Span<WSABuffer> wsaBuffers = count <= MaxStackWSABuffers ? stackalloc WSABuffer[count] : array.AsSpan(0, count);
```

Comment:
If you care more for removing the duplicate comparison of `count` and `MaxStackWSABuffers` than a bit more convoluted code, it could be reduced to
```suggestion
                WSABuffer[]? array = null;
                Span<WSABuffer> wsaBuffers = count <= MaxStackWSABuffers 
                    ? stackalloc WSABuffer[count] 
                    : (array = ArrayPool<WSABuffer>.Shared.Rent(count)).AsSpan(0, count);
```